### DISCUSSION TITLE
Build failed ruby-build 20211227-3-gcdc215e  Mac OS X 10.9.5

### DISCUSSION THREAD

**Discussion by dvodvo** [rbenv/ruby-build#1908] [Build failures] [answered]
*note: I realise there is a legacy system inovlved, however I have hit the same error on two systems, the other a much more recent one*

Initially, I started with `rbenv install 3.1.0`  & was directed to calling `3.1.0-dev` as being an available version.
Open ssl installed `Installed openssl-1.1.1i` ... but then 
```
Installing ruby-master...
ruby-build: using readline from homebrew
```
failed as it relies on homebrew and it is not upgradable for this system.

So I then went to update ruby-build: `git -C "$(rbenv root)"/plugins/ruby-build pull`   
and then upon completion: `rbenv install 3.1.0`

This time openssl 1.1.1l  Failed
```
Installing openssl-1.1.1l...

BUILD FAILED (Mac OS X 10.9.5 using ruby-build 20211227-3-gcdc215e)
```
I find it relevant that on OS X 10.13.6, running `brew update && brew upgrade ruby-build` completes with some issues raised, notably: :the following dependents of upgraded formulae are not upgraded because they are not bottled, including attention getting: 
```
geos
postgis
postgresql
rbenv
```
launching `rbenv install 3.1.0`  fails invoking ruby-build 20211227  during installation of openssl 1.1.1l, with an identical reference, however the logs are not exactly identical.  They are enclosed and the earlier time stamp refers to the older of the two systems mentioned.
```
make[1]: *** [crypto/rand/rand_unix.o] Error 1
make[1]: *** Waiting for unfinished jobs....
```
[ruby-build.20220113124112.65442.log](https://github.com/rbenv/ruby-build/files/7862546/ruby-build.20220113124112.65442.log)
[ruby-build.20220113124728.78206.log](https://github.com/rbenv/ruby-build/files/7862574/ruby-build.20220113124728.78206.log)

Interestingly without upgrading ruby-build, openssl would install.  With the upgrade (manually, or with homebrew) it would not.  This may not be a ruby-build direct issue, but sounds like a dependency matter.

How can one step through each build step, possibly manually?

**Comment by dvodvo** [comment, marked as answer]
This gets more disconcerting...  On 10.13.6  `brew install openssl@1.1`  installs 1.1.1m (one step past the 'l' version)

However, that does not change the behaviour of 'rbenv' installation process which ignores the `usr/local/Cellar` directory used by homebrew.
```
rbenv install 3.1.0
Downloading openssl-1.1.1l.tar.gz...
```
*leading to the same failure*
```
Last 10 log lines:
/usr/include/CommonCrypto/CommonRandom.h:35:9: error: unknown type name 'CCCryptorStatus'
typedef CCCryptorStatus CCRNGStatus;
        ^
crypto/rand/rand_unix.c:385:47: error: use of undeclared identifier 'kCCSuccess'
    if (CCRandomGenerateBytes(buf, buflen) == kCCSuccess)
                                              ^
2 errors generated.
make[1]: *** [crypto/rand/rand_unix.o] Error 1
```

Can ruby-build be directed to use the homebrew installed version of openssl and get on with the process?

**Reply by mislav** [reply]
You can link to Homebrew's openssl like so:
```
RUBY_CONFIGURE_OPTS="--with-openssl-dir=$(brew --prefix openssl@1.1)" rbenv install 3.1.0
```

**Reply by dvodvo** [reply]
Ah! I guess I threw myself off in the mention of brew awareness in addition to not paying proper attention to the wiki. 

As this problem is raised frequently in the discussions [this will shortcircuit matters](https://github.com/rbenv/ruby-build)
